install/kubernetes/cilium/charts/operator/templates/deployment.yaml

```diff
@@ -54,6 +54,9 @@ spec:
       - args:
         - --config-dir=/tmp/cilium/config-map
         - --debug=$(CILIUM_DEBUG)
+{{- if .Values.global.azure.userAssignedIdentityID }}
+        - --azure-user-assigned-identity-id={{ .Values.global.azure.userAssignedIdentityID }}
+{{- end }}
```

Comment:
This change is currently implemented in the deployment, but the `--config-dir` option above ensures that the `cilium-config` configmap options will be made available to the operator. You should be able to implement this instead by adding a new key/value to the configmap with the name `azure-user-assigned-identity-id` instead, which will keep the deployment templates here separate from where all the config options are declared (currently `install/kubernetes/cilium/charts/config/templates/configmap.yaml`).

Additionally, to let you know, PR #13259 is about to do some significant refactoring of the files that declare these settings to avoid declaring all options as `global`. We anticipate to merge that PR this week, in which case the changes for this PR would need to apply to a different file location.